scripts/check_compliance.py

```diff
@@ -606,43 +618,57 @@ def report_to_github(repo, pull_request, sha, suite, docs):
             if case.name not in ['Gitlint', 'Identity/Emails', 'License']:
                 comment += "```\n"
 
-            commit.create_status('failure',
-                                 docs[case.name],
-                                 'Checks failed',
-                                 '{}'.format(case.name))
+            if commit:
+                commit.create_status('failure',
+                                     docs[case.name],
+                                     'Checks failed',
+                                     '{}'.format(case.name))
         elif case.result.type in ['error']:
-            print("reporting error on %s" %case.name)
-            commit.create_status('error',
-                                 docs[case.name],
-                                 'Error during verification, please report!',
-                                 '{}'.format(case.name))
+            error("  - reporting error on %s" %case.name)
+            if commit:
+                commit.create_status('error',
+                                     docs[case.name],
+                                     'Error during verification, please report!',
+                                     '{}'.format(case.name))
         else:
-            print("Unhandled status")
+            warning("  - Unhandled status")
 
 
-    if not repo and not pull_request:
+    if not repo and not pull_request and not dryrun:
         return comment_count
 
-    if comment_count > 0:
-        comments = gh_pr.get_issue_comments()
-        commented = False
-        for cmnt in comments:
-            if ('Found the following issues, please fix and resubmit' in cmnt.body or
-                '**All checks are passing now.**' in cmnt.body) and cmnt.user.login == username:
-                if cmnt.body != comment:
-                    cmnt.edit(comment)
-                commented = True
-                break
-
-        if not commented:
-            gh_pr.create_issue_comment(comment)
+    if dryrun:
+        info("\nComment to be posted on Github:\n")
+        info("-----------------------------------")
+        info(comment)
+        info("-----------------------------------")
     else:
-        comments = gh_pr.get_issue_comments()
-        for cmnt in comments:
-            if 'Found the following issues, please fix and resubmit' in cmnt.body and cmnt.user.login == username:
-                cmnt.edit("**All checks are passing now.**\n\nReview history of this comment for details about previous failed status.\n"
-                          "Note that some checks might have not completed yet.")
-                break
+        if comment_count > 0:
```

Comment:
It's very surprising to already know the comment_count even *before* getting the comments! I suspect some renaming like `checks_new_comment`, `new_comments_count` and `all_old_comments` would help a lot.